packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -0,0 +1,404 @@
+        </div >
+        { labelPosition === Position.bottom && <div className='ms-labelWrapper' style={ this._labelDirectionHelper() }>
+          { icon && <i className={ css('ms-SpinButtonIcon', 'ms-Icon', ('ms-Icon--' + icon)) } aria-hidden='true'></i> }
+          { label &&
+            <Label
+              id={ this._labelId }
+              htmlFor={ this._inputId }
+              className='ms-SpinButtonLabel'>{ label }
+            </Label>
+          }
+        </div> }
+      </ div >
+    ) as React.ReactElement<{}>;
+  }
+
+  /**
+   * OnFocus select the contents of the input
+   */
+  public focus() {
+    if (this.state.spinning || this.state.keyboardSpinDirection !== 0) {
+      this._stop();
+    }
+
+    this._input.focus();
+    this._input.select();
+  }
+
+  /**
+   * Validate function to use if one is not passed in
+   */
+  private _defaultOnValidate = (value: string) => {
+    if (isNaN(+value)) {
+      return this._lastValidValue;
+    }
+    const newValue = Math.min(this.props.max, Math.max(this.props.min, +value));
+    return String(newValue);
+  }
+
+  /**
+   * Increment function to use if one is not passed in
+   */
+  private _defaultOnIncrement = (value: string) => {
+    let newValue = Math.min(+value + this.props.step, this.props.max);
+    return String(newValue);
+  }
+
+  /**
+   * Increment function to use if one is not passed in
+   */
+  private _defaultOnDecrement = (value: string) => {
+    let newValue = Math.max(+value - this.props.step, this.props.min);
+    return String(newValue);
+  }
+
+  /**
+   * Set the label direction with the gap on the correct side
+   */
+  private _labelDirectionHelper(): any {
+    let style: any = {};
+
+    switch (this.props.labelPosition) {
+      case Position.start:
+        style = { float: 'left', marginRight: this.props.labelGapSpace + 'px' };
+        break;
+      case Position.end:
+        style = { float: 'right', marginLeft: this.props.labelGapSpace + 'px' };
+        break;
+      case Position.top:
+        style = { marginBottom: this.props.labelGapSpace + 'px' };
+        break;
+      case Position.bottom:
+        style = { marginTop: this.props.labelGapSpace + 'px' };
+    }
+
+    return style;
+  }
+
+  private _onChange() {
+    /**
+     * A noop input change handler.
+     * https://github.com/facebook/react/issues/7027.
+     * Using the native onInput handler fixes the issue but onChange
+     * still need to be wired to avoid React console errors
+     * TODO: Check if issue is resolved when React 16 is available.
+     */
+  }
+
+  /**
+   * This is used when validating text entry
+   * in the input (not when changed via the buttons)
+   * @param event - the event that fired
+   */
+  private _validate(event: React.FocusEvent<HTMLInputElement>) {
+    const element: HTMLInputElement = event.target as HTMLInputElement;
+    const value: string = element.value;
+    if (this.state.value) {
+      const newValue = this._onValidate(value);
```

Comment:
We just updated this function, so that it can be used just for validating (without needing to return a new value).